<?xml version="1.0"?>
<launch>
	<!-- ros_args.launch -->
  	<arg name="world_name_myworld" default="empty_world"/>

	<!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  	<include file="$(find gazebo_ros)/launch/$(arg world_name_myworld).launch">
    	<!-- more default parameters can be changed here -->
		<arg name="world_name" value="$(find nuturtlebot)/worlds/block.world"/>
  	</include>

	<!-- Convert an xacro and put on parameter server -->
  	<rosparam file="$(find nuturtle_description)/config/diff_params.yaml" command="load"/>
	<param name="robot_description" command="$(find xacro)/xacro --inorder $(find nuturtle_gazebo)/xacro/diff_drive.gazebo.xacro"/>

	<!-- Spawn a robot into Gazebo -->
	<node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model turtlebot3" />
</launch>
